### PR TITLE
Add ResourceSync resource list capability

### PR DESCRIPTION
**Pull request by jcoyne**
@projecthydra/sufia-code-reviewers

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 inherit_from: .rubocop_todo.yml
 
 AllCops:
+  TargetRubyVersion: 2.1
```
Comment (awead) [inline .rubocop.yml]: Why target 2.1 when we're testing on 2.2?

File: lib/sufia/resource_sync/resource_list_writer.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            build_resource(xml, doc)
+          end
+        end
+
+        def build_resource(xml, doc)
+          xml.url do
+            key = doc.fetch('has_model_ssim', []).first.constantize.model_name.singular_route_key
+            xml.loc routes.send(key + "_url", doc['id'], host: resource_host)
+            xml.lastmod doc['system_modified_dtsi']
+            # xml['rs'].md hash: '', length: '', type: ''
```
Comment (awead) [inline lib/sufia/resource_sync/resource_list_writer.rb]: remove?

File: app/controllers/sufia/resource_sync_controller.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+class Sufia::ResourceSyncController < ApplicationController
+  def source_description
+    host = request.host
+    # Caching based on host, for multitenancy support
+    body = Rails.cache.fetch("source_description_#{host}", expires_in: 1.week) do
```
Comment (awead) [inline app/controllers/sufia/resource_sync_controller.rb]: Does using the local `host` variable do anything versus just calling `request.host` directly, or is that related to the cache?

File: .rubocop.yml (hunk 1 above)
Comment (jcoyne) [inline .rubocop.yml]: It's a language spec, not an implementation spec. I just bumped it from 2.0

File: app/controllers/sufia/resource_sync_controller.rb (hunk 3 above)
Comment (jcoyne) [inline app/controllers/sufia/resource_sync_controller.rb]: no, nothing directly.

File: .rubocop.yml (hunk 1 above)
Comment (awead) [inline .rubocop.yml]: Then why not bump to 2.2 to match Travis?

File: app/controllers/sufia/resource_sync_controller.rb (hunk 3 above)
Comment (awead) [inline app/controllers/sufia/resource_sync_controller.rb]: then you could remove the `host` variable and use `"source_description_#{request.host}"`

File: .rubocop.yml (hunk 1 above)
Comment (jcoyne) [inline .rubocop.yml]: Travis is testing an implemented version of Ruby.  This is the spec for the language spec. They aren't the same thing. We aren't using any language features of the 2.2+ ruby language spec. This calls that out.

File: app/controllers/sufia/resource_sync_controller.rb (hunk 3 above)
Comment (jcoyne) [inline app/controllers/sufia/resource_sync_controller.rb]: Yes I could. What's the benefit?

Comment (awead) [inline app/controllers/sufia/resource_sync_controller.rb]: fewer lines of code to read and execute.

Comment (jcoyne) [inline app/controllers/sufia/resource_sync_controller.rb]: There's a reason we don't write all the code on one line (we could), it's so that it's more readable.

Comment (awead) [inline app/controllers/sufia/resource_sync_controller.rb]: Yes, that's true, but I don't think assigning the extra variable makes it more readable. It think it's the opposite, in fact, because now I have to mentally keep track of `host`. Whereas using the provided `request` object seems to be the Rails way of using what's provided to us, and doesn't extend the line to any extraordinary length beyond what it already is.

You also do this three times in the class, once for each public method, which to me just yells "refactor."

**Comment by awead** [comment]
@jcoyne thanks for the edit. Feel free to bust my chops on my next PR.

**Comment by jcoyne** [comment]
https://media.giphy.com/media/10rsLtGrOcCR0s/giphy.gif
